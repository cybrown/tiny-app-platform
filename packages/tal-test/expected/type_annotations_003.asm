main()
  entry:
    Literal           "a"
    Literal           32
    Literal           "b"
    Literal           "4"
    Literal           2
    MakeRecord
    DeclareLocal      name: i1, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal           "a"
    Literal           32
    Literal           "b"
    Literal           "4"
    Literal           "c"
    Literal           3
    Literal           3
    MakeRecord
    DeclareLocal      name: i2, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal           "a"
    Literal           32
    Literal           1
    MakeRecord
    DeclareLocal      name: i3, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal           "a"
    Literal           32
    Literal           "b"
    Literal           0
    Literal           2
    MakeRecord
    DeclareLocal      name: i4, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal           "b"
    Literal           "4"
    Literal           "c"
    Literal           3
    Literal           2
    MakeRecord
    DeclareLocal      name: i5, mutable: true, hasInitialValue: true
