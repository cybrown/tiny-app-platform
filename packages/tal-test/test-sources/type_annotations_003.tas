// same fields => OK
var i1: {a: number, b: string} = {a: 32, b: "4"}

// extra field => OK
var i2: {a: number, b: string} = {a: 32, b: "4", c: 3}

// missing field => KO
var i3: {a: number, b: string} = {a: 32}

// wrong type field => KO
var i4: {a: number, b: string} = {a: 32, b: 0}

// missing nullable field and extra field => OK
var i5: {a: number | null, b: string} = {b: "4", c: 3}
