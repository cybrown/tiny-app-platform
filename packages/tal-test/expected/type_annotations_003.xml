<program>
  <comment>
    <declarelocal>
      <record>
        <recordentry>
          <literal />
        </recordentry>
        <recordentry>
          <literal />
        </recordentry>
      </record>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <record>
        <recordentry>
          <literal />
        </recordentry>
        <recordentry>
          <literal />
        </recordentry>
        <recordentry>
          <literal />
        </recordentry>
      </record>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <record>
        <recordentry>
          <literal />
        </recordentry>
      </record>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <record>
        <recordentry>
          <literal />
        </recordentry>
        <recordentry>
          <literal />
        </recordentry>
      </record>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <record>
        <recordentry>
          <literal />
        </recordentry>
        <recordentry>
          <literal />
        </recordentry>
      </record>
    </declarelocal>
  </comment>
</program>
